sshd-common/src/main/java/org/apache/sshd/common/util/security/SecurityUtils.java

```diff
@@ -606,99 +609,155 @@ public static RandomFactory getRandomFactory() {
      * @return {@code true} if EDDSA curves (e.g., {@code ed25519}) are supported
      */
     public static boolean isEDDSACurveSupported() {
+        return getEdDSASupport().isPresent();
+    }
+
+    public static boolean isNetI2pCryptoEdDSARegistered() {
+        register();
+        return isProviderRegistered(EDDSA);
+    }
+
+    public static Optional<EdDSASupport<?, ?>> getEdDSASupport() {
         if (isFipsMode()) {
-            return false;
+            return Optional.empty();
```

Comment:
I think this is incorrect as before there was only a single EDSA provider which was never FIPS compliant, now there is a FIPS compliant (bouncycastle) one (when using the FIPS variant) so it should be up to the provider to perform the `isFipsMode` check and return an empty optional if their EDDSA support is not compliant.

Reply:
Not sure. I don't see EdDSA listed as an "approved algorithm" in the [BC FIPS Security Policy 2.0.0](https://downloads.bouncycastle.org/fips-java/docs/BC-FJA-SecurityPolicy-2.0.0.pdf). It's still listed in "Table 6 - Non-Approved Algorithms Not Allowed in the Approved Mode of Operation".

Reply:
my mistake I am confusing ECDSA and EDDSA again :-(

Reply:
OK. I still see the point, though. A future BC-FIPS might have EdDSA as "approved algorithm". It _is_ in [NIST SP 800-140C Rev. 2](https://csrc.nist.gov/pubs/sp/800/140/c/r2/final) from 2023-07-25, and is listed at [SP 800-140C: Approved Security Functions](https://csrc.nist.gov/projects/cmvp/sp800-140c).

And BC currently has BC FIPS 2.1.0 undergoing validation. So we might have to refactor this approach in the near future.